### ISSUE TITLE
Feature request: automated publishing to Docker Hub

### ISSUE THREAD

**Issue by taeh98** [Tribler/tribler#6591]
Hello, I hope this finds you well.

Please could you add an action to this repository that would automatically make it publish new releases to Docker Hub? The availability of an image on Docker Hub allows for greater portability and ease with which to use and extend Tribler (for example, to run Tribler in a container and interact with it via its API). I see you have [existing images on Docker Hub](https://hub.docker.com/u/tribler), but all of them are several years old and out of date. By making an action to publish an image to Docker Hub automatically, this problem would be solved. It would remove the overhead of manually publishing images with each release (or alternatively not doing so and the image becoming out of date). The following links give some more context as to how this could be achieved:

- https://docs.github.com/en/actions/publishing-packages/publishing-docker-images
- https://github.com/marketplace/actions/build-and-push-docker-images
- https://docs.docker.com/ci-cd/github-actions

Thank you for your time and consideration.

**Comment by ichorid** [comment]
@xoriole , when was the last time we built Docker image? Why did we stop to do so? Was it too hard to maintain?

**Comment by qstokkink** [comment]
Small 2024 update: PR #8108 introduces automatic publishing to GitHub Packages.

**Comment by qstokkink** [comment]
The latest main version of Tribler is now available here: https://github.com/Tribler/tribler/pkgs/container/tribler

Actually pushing to Docker hub is still TODO but that should not be too much additional work.

**Comment by jkaberg** [comment]
@qstokkink this doesn't seem to work properly? None of the [8 series are pushed](https://github.com/Tribler/tribler/releases) as containers. An optimal setup would build and push stable releases to the :latest tag but also versioned :tags would be cool. Quite common to have some kind of :dev tag for bleeding edge as well. 

Also it would be nice with some docs on how to use the containers, especially now that the 8 series are released with the headless setup and web ui.

**Comment by qstokkink** [comment]
You are right: currently, the Docker GitHub package is "barely functional" and stuck to the bleeding edge version. Getting everything packed with proper builds and tags, is something I plan to get done for the 8.1.0 release. I also want to bring back `compose`.

**Comment by qstokkink** [comment]
For 8.1.0 I'll at least get a "stable" release up and running. We handle the remainder of this issue once we gain control over our own account again then.

**Comment by qstokkink** [comment]
Alright, we now have a "stable" version living in `docker pull ghcr.io/tribler/tribler:latest`, instead of the bleeding edge experimental code.

I'll postpone the remainder of this issue, automatically publishing to Docker hub, until `8.2.0` milestone.

**Comment by qstokkink** [comment]
I contacted Docker Hub support to try and reclaim `u/tribler`. We'll see what happens.
